### PR TITLE
Use new runner authentication token

### PR DESCRIPTION
**Pull request by mx-moth**
Introduced in Gitlab 15, required from Gitlab 17.

### CODE REVIEW

File: register_and_run.sh
Comment (sharon-tickell) [inline register_and_run.sh:14]: Good call on the redaction.... but could this  be [redacted] if the token is set or [empty] if the token has not been set? The latter strikes me as a likely error case that it would be good to have some debugging help with

File: start_local_runner.sh
Comment (sharon-tickell) [inline start_local_runner.sh:72]: I wonder if there is a way to avoid putting the token in plain text on the command line? Perhaps something with secrets?   Though that may then be very swarm-specific, and more trouble than it's worth.
(Also we've always had the token on the command line before, so no need to hold up merging on account of this question)

**Review by sharon-tickell — APPROVED** [review]
Looks good: a couple of suggestions from me, but nothing to hold up merging once you're ready to be able to configure these

File: register_and_run.sh
Comment (mx-moth) [reply register_and_run.sh:14]: A few lines above this, an error is thrown if $TOKEN is unset or empty, so this guard is already taken care of.

File: start_local_runner.sh
Comment (mx-moth) [reply start_local_runner.sh:72]: It would be pretty trivial to read the TOKEN from a secret file mounted in to the container, however I don't think this would noticeably increase security. Regardless of how securely we ended up getting the TOKEN value in to a container, each container is mounted with write access to the hosts docker socket. Every runner would be able to discover the token of any other runner by doing a `docker inspect <other-runner>` and following the information there. The runners will start other containers as part of CI, but these other containers won't inherit the environment variables or the secrets. The other containers may have a docker socket mounted presenting the same `docker inspect` hole as the runner.

Comment (mx-moth) [reply start_local_runner.sh:72]: And specifically with this `start_local_runner.sh` script, it is explicitly for debugging pipeline issues and testing changes to the runner image, not for production use. Having a token in the command line on your local development machine is less of an issue - if you can't trust your development environment then you've got bigger issues!

Comment (sharon-tickell) [reply start_local_runner.sh:72]: All good - I agree with you on the relative security

**Pull request merged**
